rclcpp/src/rclcpp/parameter_client.cpp

```diff
@@ -144,6 +144,40 @@ AsyncParametersClient::get_parameters(
   return future_result;
 }
 
+std::shared_future<std::vector<rcl_interfaces::msg::ParameterDescriptor>>
+AsyncParametersClient::describe_parameters(
+  const std::vector<std::string> & names,
+  std::function<
+    void(std::shared_future<std::vector<rcl_interfaces::msg::ParameterDescriptor>>)
+  > callback)
+{
+  auto promise_result =
+    std::make_shared<std::promise<std::vector<rcl_interfaces::msg::ParameterDescriptor>>>();
+  auto future_result = promise_result->get_future().share();
+
+  auto request = std::make_shared<rcl_interfaces::srv::DescribeParameters::Request>();
+  request->names = names;
+
+  describe_parameters_client_->async_send_request(
+    request,
+    [promise_result, future_result, callback](
+      rclcpp::Client<rcl_interfaces::srv::DescribeParameters>::SharedFuture cb_f)
+    {
+      std::vector<rcl_interfaces::msg::ParameterDescriptor> descriptors;
+      std::vector<rcl_interfaces::msg::ParameterDescriptor> & descs = cb_f.get()->descriptors;
+      for (rcl_interfaces::msg::ParameterDescriptor & desc : descs) {
+        descriptors.push_back(desc);
+      }
+      promise_result->set_value(descriptors);
```

Comment:
Looking at this again, do we actually need to do this?  Isn't this just copying the data from one vector into another?  Instead, can we just do:

```
promise_result->set_value(cb_f.get()->descriptors);
```

(one thing I don't know about is the lifetime of the objects, so that bears thinking about)

Reply:
Nice catch 👍 thanks for the suggestion!